subsys/fs/ext2/ext2_impl.h

```diff
@@ -0,0 +1,342 @@
+
+int assign_block_num(struct ext2_data *fs, struct block *b);
+
+int init_blocks_slab(struct ext2_data *fs);
+
+/* FS operations */
+
+/**
+ * @brief Initialize structure with data needed to access the storage device
+ *
+ * @param fs File system data structure to initialize
+ * @param storage_dev Pointer to storage
+ * @param flags Additional flags (e.g. RO flag)
+ *
+ * @retval 0 on success
+ * @retval -EINVAL when superblock of ext2 was not detected
+ * @retval -ENOTSUP when described file system is not supported
+ * @retval <0 other error
+ */
+int ext2_init_storage(struct ext2_data **fsp, const void *storage_dev, int flags);
+
+/**
+ * @brief Verify superblock of file system
+ *
+ * Checks if file system is supported by the implementation.
+ * @retval 0 when superblock is valid
+ * @retval -EROFS when superblock is not valid but file system may be mounted read only
+ * @retval -EINVAL when superblock is not valid and file system cannot be mounted at all
+ * @retval -ENOTSUP when superblock has some field set to value that we don't support
+ */
+int ext2_verify_superblock(struct disk_superblock *sb);
+
+/**
+ * @brief Initialize all data needed to perform operations on file system
+ *
+ * Fetches the superblock. Initializes structure fields.
+ */
+int ext2_init_fs(struct ext2_data *fs);
+
+/**
+ * @brief Clear the data used by file system implementation
+ *
+ */
+int ext2_close_fs(struct ext2_data *fs);
+
+/**
+ * @brief Clear the data used to communicate with storage device
+ *
+ */
+int ext2_close_struct(struct ext2_data *fs);
+
+/**
+ * @brief Create the ext2 file system
+ *
+ * This function uses functions stored in `ext2_data` structure to create new
+ * file system on storage device.
+ *
+ * NOTICE: fs structure must be first initialized with `ext2_init_fs` function.
+ *
+ * After this function succeeds the `ext2_clean` function must be called.
+ *
+ * @param fs File system data (must be initialized before)
+ *
+ * @retval 0 on success
+ * @retval -ENOSPC when storage device is too small for ext2 file system
+ * @retval -ENOTSUP when storage device is too big (file systems with more than
+ *         8192 blocks are not supported)
+ */
+int ext2_format(struct ext2_data *fs, struct ext2_cfg *cfg);
+
+/* Lookup flags */
+#define LOOKUP_ARG_OPEN BIT(0)
+#define LOOKUP_ARG_CREATE BIT(1)
+#define LOOKUP_ARG_STAT BIT(2)
+#define LOOKUP_ARG_UNLINK BIT(3)
+
+/* Structure for lookup arguments and results.
+ *
+ * Required fields (must be filled when lookup function is invoked):
+ *  - path
+ *  - flags
+ *
+ * Fields that hold the result:
+ *  - inode
+ *  - parent
+ *  - offset
+ *  - name_pos
+ *  - name_len
+ *
+ * Some of these fields have a meaning only for a specific function.
+ * (E.g. during stat only the fields parent and offset are used)
+ *
+ * Field is marked with these labels when lookup is used from other function:
+ *  OP -- open
+ *  CR -- create
+ *  ST -- stat
+ *  UN -- unlink
+ */
+struct lookup_args {
```

Comment:
As previous comment, "lookup_args" without a prefix seems like it could have a conflict with other libraries or code

Reply:
done.